gsy_framework/data_classes.py

```diff
@@ -173,9 +174,6 @@ def from_dict(offer: Dict) -> "Offer":
 
     def __eq__(self, other) -> bool:
         return (self.id == other.id and
-                self.price == other.price and
-                self.original_price == other.original_price and
-                self.energy == other.energy and
```

Comment:
I removed these in order to not fall into the trap of comparing `0.99999999` and `10` e.g. for the price. The comparison of the `id` should suffice in general, but I am OK to keep the other checks here too.